zebra-rpc/src/methods.rs

```diff
@@ -647,9 +647,22 @@ where
 
     // TODO: use a generic error constructor (#5548)
     fn get_raw_mempool(&self) -> BoxFuture<Result<Vec<String>>> {
+        use zebra_chain::block::MAX_BLOCK_BYTES;
+
+        /// Determines whether the output of this RPC is sorted like zcashd
+        const SHOULD_USE_ZCASHD_ORDER: bool = true;
+
         let mut mempool = self.mempool.clone();
 
         async move {
+            #[cfg(feature = "getblocktemplate-rpcs")]
+            let request = if SHOULD_USE_ZCASHD_ORDER {
+                mempool::Request::FullTransactions
+            } else {
+                mempool::Request::TransactionIds
+            };
+
+            #[cfg(not(feature = "getblocktemplate-rpcs"))]
             let request = mempool::Request::TransactionIds;
```

Comment:
Optional: this might be a bit clearer with `cfg!()`:
https://doc.rust-lang.org/std/macro.cfg.html

```suggestion
            let request = if SHOULD_USE_ZCASHD_ORDER && cfg!(feature = "getblocktemplate-rpcs") {
                mempool::Request::FullTransactions
            } else {
                mempool::Request::TransactionIds
            };
```

Reply:
Skipping/reverting this one because the compiler didn't like it without the feature flag:

>    --> zebra-rpc/src/methods.rs:660:35
>     |
> 660 |                 mempool::Request::FullTransactions
>     |                                   ^^^^^^^^^^^^^^^^ variant or associated item not found in `zebra_node_services::mempool::Request`

https://github.com/ZcashFoundation/zebra/actions/runs/3955201675/jobs/6773274036#step:10:377